pkg/agent/controller/endpointslice.go

```diff
@@ -0,0 +1,228 @@
+package controller
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/util/retry"
+
+	discovery "k8s.io/api/discovery/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/workqueue"
+)
+
+func NewEndpointSliceController(spec *AgentSpecification, cfg *rest.Config) (*EndpointSliceController, error) {
+	kubeClientSet, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("Error building clientset: %s", err.Error())
+	}
+
+	endpointSliceController := EndpointSliceController{
+		endPointSlicequeue: workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		kubeClientSet:      kubeClientSet,
+		clusterID:          spec.ClusterID,
+		namespace:          spec.Namespace,
+	}
+
+	return &endpointSliceController, nil
+}
+
+func (e *EndpointSliceController) Start(stopCh <-chan struct{}) error {
+	e.endpointSliceInformer = cache.NewSharedIndexInformer(
+		&cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				return e.kubeClientSet.DiscoveryV1beta1().EndpointSlices(e.namespace).List(options)
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				return e.kubeClientSet.DiscoveryV1beta1().EndpointSlices(e.namespace).Watch(options)
+			},
+		},
+		&discovery.EndpointSlice{},
+		0,
+		cache.Indexers{},
+	)
```

Comment:
Use the generated clientset informer code?

Reply:
Removed the controller.